src/core.rs

```diff
@@ -197,6 +206,14 @@ impl Core {
         Ok(init_result)
     }
 
+    pub fn get_system_view(&self) -> Option<Arc<BarrierView>> {
+        self.system_view.clone()
```

Comment:
In what case, `system_view` could be None?

Reply:
> In what case, `system_view` could be None?

this case:
```
let core = Core {
    system_view: None,
    //...
};
```